<launch>
    <arg name="sim" default="true"/>
    <arg name="hardware_connected" default="false"/>
    <arg name="enable_reference_pose" default="false"/>
    <arg name="run_rviz" default="true"/>
    <arg name="run_rqt_multiplot" default="false"/>

    <node pkg="oct_levitation" name="single_dipole_x_control" type="x_control_single_dipole_full_alloc.py" output="screen"/>

    <group if="$(arg enable_reference_pose)">
        <node pkg="oct_levitation" name="reference_pose_publisher" type="refrence_pose_publisher.py" output="screen"/>
    </group>

    <group if="$(arg sim)">
        <node pkg="oct_levitation" name="free_body_simulation" type="free_rigid_body_simulator.py" output="screen"/>
        <node pkg="oct_levitation" name="control_input_delaying_node" type="control_input_delay_sim.py" output="screen">
            <param name="delay_time" value="0.0" />
        </node>
    </group>

    <group if="$(eval not sim and hardware_connected)">
        <node pkg="tnb_mns_driver" name="tnb_mns_driver_node" type="tnb_mns_driver_node" output="screen" launch-prefix="xterm -e"/>
    </group>

    <group if="$(arg run_rviz)">
        <node pkg="oct_levitation" name="rviz_visualizations_node" type="rviz_visualizations.py" output="screen"/>
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(find oct_levitation)/config/rviz/single_dipole_control_config.rviz" output="screen"/>
    </group>

    <group if="$(arg run_rqt_multiplot)">
        <node pkg="rqt_multiplot" type="rqt_multiplot" name="rqt_multiplot" output="screen"
            args="--force-config $(find your_package)/config/rqt_multiplot/rqt_multiplot_pose_config.xml" />
    </group>

</launch>